lib/internal/modules/esm/fetch_module.js

```diff
@@ -99,118 +96,75 @@ function fetchWithRedirects(parsed) {
     return existing;
   }
   const handler = parsed.protocol === 'http:' ? HTTPGet : HTTPSGet;
-  const result = new Promise((fulfill, reject) => {
+  const result = (async () => {
     const req = handler(parsed, {
-      headers: {
-        Accept: '*/*'
-      }
-    })
-    .on('error', reject)
-    .on('response', (res) => {
-      function dispose() {
-        req.destroy();
-        res.destroy();
-      }
+      headers: { Accept: '*/*' },
+    });
+    const { 0: res } = await once(req, 'response');
+    try {
       if (res.statusCode >= 300 && res.statusCode <= 303) {
-        if (res.headers.location) {
-          dispose();
-          try {
-            const location = new URL(res.headers.location, parsed);
-            if (location.protocol !== 'http:' &&
-              location.protocol !== 'https:') {
-              reject(new ERR_NETWORK_IMPORT_DISALLOWED(
-                res.headers.location,
-                parsed.href,
-                'cannot redirect to non-network location'));
-              return;
-            }
-            return PromisePrototypeThen(
-              PromiseResolve(fetchWithRedirects(location)),
-              (entry) => {
-                cacheForGET.set(parsed.href, entry);
-                fulfill(entry);
-              });
-          } catch (e) {
-            dispose();
-            reject(e);
-          }
+        const location = new URL(res.headers.location, parsed);
```

Comment:
not all of these will have a location header, should be checked (idc on behavior after check)

Reply:
I'm kind of not sure about that behavior - the old behavior if I understand correctly was to just ignore it if there is no location header. I'll preserve this behavior for now but lmk if you have another preference?